ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendUtils.java

```diff
@@ -133,4 +142,33 @@ static RestRequestMetricsGroup getMetricsGroupForGet(FrontendMetrics frontendMet
     }
     return group;
   }
+
+  /**
+   * Refresh the OperationOrBlobId in request path with the specific latest version info.
+   * @param restRequest the {@link RestRequest}
+   * @param datasetVersionRecord the {@link DatasetVersionRecord} which include the version info.
+   * @throws RestServiceException
+   */
+  static void refreshRequestPathWithNewOperationOrBlobIdIfNeeded(RestRequest restRequest,
+      DatasetVersionRecord datasetVersionRecord, String version) throws RestServiceException {
+    if (version.equals("LATEST") || version.equals("MAJOR") || version.equals("MINOR") || version.equals("PATCH")) {
+      RequestPath originalRequestPath = (RequestPath) restRequest.getArgs().get(REQUEST_PATH);
+      String originalOperationOrBlobId = originalRequestPath.getOperationOrBlobId(false);
```

Comment:
please pass true to this function call, otherwise, if the operationOrBlobId is "/dataset" then the latestOperationOrBlobId would become "/version"

Reply:
The originalOperationOrBlobId will follow the format like: /named/account/container/dataset/version since I verified in setTargetDatasetAndVersionInRestRequestIfNeeded. This issue won't happen.